ReplayTables/ReplayBuffer.py

```diff
@@ -48,6 +49,28 @@ def sample(self, n: int) -> Batch:
         samples = self._storage.get(idxs)
         return samples
 
+    def sample_without_replacement(self, n: int) -> Batch:
+        # most of the time, we get unique idxs in the first sample
+        # so we fastpath past the type conversions and set additions
+        # for that common case for performance reasons.
+        sub_idxs = self._sampler.sample(n)
+        uniq_idxs = set(sub_idxs)
+
+        if len(uniq_idxs) < n:
+            for _ in range(25):
```

Comment:
Ideally we'd make this magic constant a user-configurable option. Don't really have great infrastructure for that right now, and this seems like a sufficiently conservative choice so ignoring it for now.